index.bs

```diff
@@ -103,27 +78,234 @@ Introduction {#intro}
 
 <section class="non-normative">
 
-This potential module describes a mechanism for allowing WebXR applications to cast rays into the users real world environment and report back, to the best of the XR device's knowledge, the point at which the ray intersected with a physical object and the orientation of the intersected surface. This allows for virtual objects to be placed in alignment with those surfaces, such as placing objects realistically on the floor or attaching them to a wall.
+This module describes a mechanism for allowing WebXR applications to cast rays into the users' real world environment and report back, to the best of the XR device's knowledge, the point at which the ray intersected with a physical object along with the orientation of the intersected surface. This allows for virtual objects to be placed in alignment with those surfaces, such as placing objects realistically on the floor or attaching them to a wall. The hit test API is an extension to <a href="https://www.w3.org/TR/webxr/">WebXR Device API</a> and builds on top of <a href="https://github.com/immersive-web/webxr-ar-module/">WebXR Augmented Reality Module</a>.
 
 </section>
 
-WebXR Device API Integration {#webxr-device-api-integration}
-============================
-This module expands the definition of {{XRSession}}.
+Terminology {#terminology}
+-----------
+
+Hit testing, as understood by this document, is an act of checking if an idealised mathematical ray (half-line) intersects with real world as understood by the underlying Augmented Reality hardware & software. Ray intersections against virtual objects created by the application consuming the API are explicitly out of scope of the hit test API.
+
+Hit test options {#hit-test-options}
+================
+
+XRHitTestTrackableType {#hit-test-trackable-type-enum}
+----------------------
+
+An {{XRHitTestTrackableType}} enum specifies the type of entity that can be used for the purposes of hit test source creation.
+
+<script type="idl">
+enum XRHitTestTrackableType {
+  "point",
+  "plane"
+};
```

Comment:
What about meshes?  Points and planes work fine for ARKit and ARCore, but as we implement this on HMDs that work with meshes, not planes, we will want to allow that.  Or is the intent that a mesh is a very large collection of very small planes?

Reply:
I'm hoping that the spec is flexible enough that when we start implementing it on HMDs & start dealing with meshes, we will only need to add one more value to this enum and add more text in "native entity type" section. As you mention, hit test against meshes could also be implemented by the UAs in a spec-compliant way by treating mesh as collection of planes, although it's probably better to extend the enum - the main reason for this enum is to guarantee some stability of behavior for the apps that already want to use the API while still allowing us to make changes to the API.

Reply:
I anticipate we'll be implementing it rather soon on an HMD, so I think it would be good for @Manishearth and/or @thetuvix to chime in on this.

I would vote to add mesh upfront, but I'd like their opinion.  As the spec is laid out, I think it would work fine with Hololens meshes (for example), since a hit would return a pose (as described above) as a point and surface orientation.

Reply:
I've added "mesh" entity type now to reduce the amount of changes needed to the text once hit test against meshes gets implemented.